skglm/gpu/solvers/jax_solver.py

```diff
@@ -0,0 +1,89 @@
+# if not set, raises an error related to CUDA linking API.
+# as recommended, setting the 'XLA_FLAGS' to bypass it.
+# side-effect: (perhaps) slow compilation time.
+import os
+os.environ['XLA_FLAGS'] = '--xla_gpu_force_compilation_parallelism=1'  # noqa
+
+import numpy as np  # noqa
+
+import jax  # noqa
+import jax.numpy as jnp  # noqa
+# set float64 as default float type.
+# if not, amplifies rounding errors.
+jax.config.update("jax_enable_x64", True)  # noqa
+
+from skglm.gpu.utils.host_utils import compute_obj, eval_opt_crit  # noqa
+
+
+class JaxSolver:
```

Comment:
In order to test the design choices, can you call this FISTAJax and make it modular (ie pass an objective function directly) ? and same for other solvers